app/models/concerns/episode_has_feeds.rb

```diff
@@ -0,0 +1,49 @@
+require "active_support/concern"
+
+module EpisodeHasFeeds
+  extend ActiveSupport::Concern
+
+  included do
+    has_many :episodes_feeds, dependent: :delete_all
+    has_many :feeds, through: :episodes_feeds
+
+    after_initialize :set_default_feeds, if: :new_record?
+    before_validation :set_default_feeds, if: :new_record?
+
+    # NOTE: this DOES NOT apply the display_episodes_count
+    scope :in_feed, ->(feed) do
+      Episode.joins(:episodes_feeds)
+        .where(episodes_feeds: {feed: feed})
+        .published_by(feed.episode_offset_seconds.to_i)
+    end
+
+    # NOTE: this DOES NOT apply the display_episodes_count
+    scope :in_default_feed, -> { joins(:feeds).where(feeds: {slug: nil}).published }
```

Comment:
It's not trivial to apply `display_episodes_count` to the `/api/v1/episodes` endpoint.  Since it includes multiple podcasts in that single paged response.  It would need to be something like:

```SQL
SELECT e.id, e.published_at
FROM (
  SELECT id, podcast_id, published_at, 
    ROW_NUMBER() OVER (
      PARTITION BY podcast_id ORDER BY published_at DESC
    ) AS row
  FROM episodes
  WHERE published_at <= NOW()
) e 
INNER JOIN feeds f ON (f.slug IS NULL AND f.podcast_id = e.podcast_id)
WHERE (display_episodes_count IS NULL OR row <= display_episodes_count)
```

That isn't exactly quick, when returning all podcasts.

Reply:
Yeah, I think that's the next thing we should decide after this PR.  What's the utility of our public endpoints?  Just to provide a JSON version of what's in public RSS?  Or something else?

Reply:
I would say to provide a json version of the public feed and individual public episodes, and I am fairly convinced it should mirror the public feed in the scope of what data it provides, but as a JSON API, and this gets us much much closer to that